js/src/components/shipping-rate-section/minimum-order-card/minimum-order-card.js

```diff
@@ -98,14 +98,14 @@ const MinimumOrderCard = ( props ) => {
 		}
 
 		/**
-		 * When there is more than one group,
-		 * we render the groups with defined threshold first.
-		 * Then, if there is a group with undefined threshold,
-		 * we render an "Add another minimum order" button last.
+		 * The group with undefined threshold. This is used
+		 * to render the "Add another minimum order" button
+		 * after all the groups with defined threshold.
```

Comment:
> * The group with undefined threshold **_could be more than one_** when there are **_multiple currencies_** in the shipping rates.
> * The `thresholdGroups` are **_the groups with defined or undefined threshold_**, except for the `emptyThresholdGroup` one.

Hey @eason9487 , thanks, I just realized this. I wanted the `thresholdGroups` to be groups with defined threshold only, to render `MinimumOrderInputControl`. I'll fix this.

Reply:
Fixed in [`e6623ed` (#1267)](https://github.com/woocommerce/google-listings-and-ads/pull/1267/commits/e6623eddf2b46f40338896e35138d91abc34f8ec).

Reply:
🚧  or 📜  I found multi-currency becomes a complex problem here: 😂 

1. Only the first undefined threshold can be added. When the country to be set is the latter, the UI becomes incomprehensible for merchants.

   https://user-images.githubusercontent.com/17420811/157227874-6a0d9fc0-e81f-47be-abc5-a9198ad07689.mp4

2. It's possible to edit countries' thresholds across multi-currency. Although the currencies won't be overwritten, the results look a bit surprising.

   https://user-images.githubusercontent.com/17420811/157228468-895c73bb-8f83-40ff-9d96-fa68123fe959.mp4

3. It's possible to edit countries' shipping rates across multi-currency. The currencies could be overwritten, but the results look much more surprising when comparing this to 2.

   https://user-images.githubusercontent.com/17420811/157228702-2714f975-1599-4aea-a3e8-76d00f55aa4d.mp4

But this is a rare edge case and probably needs more discussion (and maybe UI design change), so I think probably we could create a follow-up issue to see how it could be solved later.

Reply:
LOL, it does. 😂 

For the issue 1, I think it is still okay. It may be tedious and difficult, but generally the final outcome is still correct.

For issue 2, that does look buggy, because users are putting "234 **AUD**" for US in the minimum order modal, but end up showing "234 **USD**" for US in the final outcome in the page. I'm thinking that can be solved if we have separate currency for the free shipping threshold, e.g.:

```js
const shippingRate = {
	id: '1',
	country: 'US',
	method: 'flat_rate',
	currency: 'USD', // currency for shipping rate.
	rate: 4.99,
	options: {
		free_shipping_threshold: 88,
		free_shipping_threshold_currency: 'AUD', // currency for free shipping threshold.
	},
}
```

However, the above idea may be too much though. 

Another idea is to simply restrict the country options. In the modal, we will only show country options that has the same AUD shipping rate currency, so country US won't show up in the drop down. The "downside" is users may be wondering for a bit "where is there no US in the drop down?" (answer: because the currency doesn't match). 

I'll create a separate issue for this. 📜 

For issue 3, although it looks a bit weird, I think that is still okay, because what the users put in the modal matches the final outcome in the page.

Reply:
> I'll create a separate issue for this. 📜

Note: I have created issue https://github.com/woocommerce/google-listings-and-ads/issues/1324 for the issue 2.